phase4-profile-entsog/src/main/java/com/helger/phase4/profile/entsog/ENTSOGPMode.java

```diff
@@ -0,0 +1,159 @@
+        eReportProcessErrorNotifyConsumer, eReportProcessErrorNotifyProducer, eReportDeliveryFailuresNotifyProducer);
+  }
+
+  @Nonnull
+  public static PModeLegSecurity generatePModeLegSecurity() {
+    final PModeLegSecurity aPModeLegSecurity = new PModeLegSecurity();
+    aPModeLegSecurity.setWSSVersion(EWSSVersion.WSS_111);
+    aPModeLegSecurity.setX509SignatureAlgorithm(ECryptoAlgorithmSign.RSA_SHA_256);
+    aPModeLegSecurity.setX509SignatureHashFunction(ECryptoAlgorithmSignDigest.DIGEST_SHA_256);
+    aPModeLegSecurity.setX509EncryptionAlgorithm(ECryptoAlgorithmCrypt.AES_128_GCM);
+    aPModeLegSecurity.setX509EncryptionMinimumStrength(Integer.valueOf(128));
+    aPModeLegSecurity.setPModeAuthorize(false);
+    aPModeLegSecurity.setSendReceipt(true);
+    aPModeLegSecurity.setSendReceiptNonRepudiation(true);
+    aPModeLegSecurity.setSendReceiptReplyPattern(EPModeSendReceiptReplyPattern.RESPONSE);
+    return aPModeLegSecurity;
+  }
+
+  @Nonnull
+  public static PModeLeg generatePModeLeg(@Nullable final String sResponderAddress) {
+    return new PModeLeg(generatePModeLegProtocol(sResponderAddress), generatePModeLegBusinessInformation(),
+        generatePModeLegErrorHandling(), (PModeLegReliability) null, generatePModeLegSecurity());
+  }
+
+  @Nonnull
+  public static PModePayloadService generatePModePayloadSevice() {
+    return new PModePayloadService(EAS4CompressionMode.GZIP);
+  }
+
+  @Nonnull
+  public static PModeReceptionAwareness generatePModeReceptionAwareness() {
+    final ETriState eReceptionAwareness = ETriState.TRUE;
+    final ETriState eRetry = ETriState.TRUE;
+    final int nMaxRetries = 1;
+    final int nRetryIntervalMS = 10_000;
+    final ETriState eDuplicateDetection = ETriState.TRUE;
+    return new PModeReceptionAwareness(eReceptionAwareness, eRetry, nMaxRetries, nRetryIntervalMS, eDuplicateDetection);
+  }
+
+  /**
+   * One-Way Version of the CEF pmode uses one-way push
+   *
+   * @param sInitiatorID      Initiator ID
+   * @param sResponderID      Responder ID
+   * @param sResponderAddress Responder URL
+   * @param aPModeIDProvider  PMode ID provider
+   * @param bPersist          <code>true</code> to persist the PMode in the
+   *                          PModeManager, <code>false</code> to have it only in
+   *                          memory.
+   * @return New PMode
+   */
+  @Nonnull
+  public static PMode createENTSOGPMode(@Nonnull @Nonempty final String sInitiatorID,
+                                        @Nonnull @Nonempty final String sResponderID,
+                                        @Nullable final String sResponderAddress,
+                                        @Nonnull final IPModeIDProvider aPModeIDProvider,
+                                        final boolean bPersist) {
+    final PModeParty aInitiator = new PModeParty("http://www.entsog.eu/eic-codes/eic-party-codes-x", sInitiatorID,
```

Comment:
Could we create a constant from `"http://www.entsog.eu/eic-codes/eic-party-codes-x"`?

Reply:
ok